python/pyspark/sql/group.py

```diff
@@ -194,6 +194,28 @@ def pivot(self, pivot_col, values=None):
             jgd = self._jgd.pivot(pivot_col, values)
         return GroupedData(jgd, self.sql_ctx)
 
+    def apply(self, udf_obj):
```

Comment:
Maybe we can define another `pandas_df_udf` for this kind of pandas udf?

Reply:
I am not sure if that is necessary. We can check if the function should have just one parameter in `apply()` without introducing a new `pandas_df_udf` too.

Reply:
I'm basically concerned that there is no distinct difference between the current pandas udf and the new one for `apply`.  But seems we can distinguish them by looking at the return type? If so, we may no need of `pandas_df_udf`.

But we should update the doc of `pandas_udf` for this kind of (`apply`) pandas udf.

Reply:
I see. It's a totally valid concern. Yeah I think we can distinguish them by returnType.

I will update the doc.

Reply:
Doc updated